presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/GenericResourceGroupConfigurationManager.java

```diff
@@ -0,0 +1,287 @@
+    {
+        return selectors;
+    }
+
+    //Copied from FileResourceGroupConfigurationManager so this logic can be reused
+    @Override
+    public void configure(ConfigurableResourceGroup group, SelectionContext context)
+    {
+        List<ResourceGroupSpec> candidates = rootGroups;
+        List<String> segments = group.getId().getSegments();
+        ResourceGroupSpec match = null;
+        for (int i = 0; i < segments.size(); i++) {
+            List<ResourceGroupSpec> nextCandidates = null;
+            ResourceGroupSpec nextCandidatesParent = null;
+            for (ResourceGroupSpec candidate : candidates) {
+                if (candidate.getName().expandTemplate(context).equals(segments.get(i))) {
+                    if (i == segments.size() - 1) {
+                        if (match != null) {
+                            throw new IllegalStateException(format("Ambiguous configuration for %s. Matches %s and %s", group.getId(), match.getName(), candidate.getName()));
+                        }
+                        match = candidate;
+                    }
+                    else {
+                        if (nextCandidatesParent != null) {
+                            throw new IllegalStateException(format("Ambiguous configuration for %s. Matches %s and %s", group.getId(), nextCandidatesParent.getName(), candidate.getName()));
+                        }
+                        nextCandidates = candidate.getSubGroups();
+                        nextCandidatesParent = candidate;
+                    }
+                }
+            }
+            if (nextCandidates == null) {
+                break;
+            }
+            candidates = nextCandidates;
+        }
+        checkState(match != null, "No matching configuration found for: %s", group.getId());
+        if (match.getSoftMemoryLimit().isPresent()) {
+            group.setSoftMemoryLimit(match.getSoftMemoryLimit().get());
+        }
+        else {
+            synchronized (generalPoolMemoryFraction) {
+                double fraction = match.getSoftMemoryLimitFraction().get();
+                generalPoolMemoryFraction.put(group, fraction);
+                group.setSoftMemoryLimit(new DataSize(generalPoolBytes * fraction, BYTE));
+            }
+        }
+        group.setMaxQueuedQueries(match.getMaxQueued());
+        group.setMaxRunningQueries(match.getMaxRunning());
+        if (match.getSchedulingPolicy().isPresent()) {
+            group.setSchedulingPolicy(match.getSchedulingPolicy().get());
+        }
+        if (match.getSchedulingWeight().isPresent()) {
+            group.setSchedulingWeight(match.getSchedulingWeight().get());
+        }
+        if (match.getJmxExport().isPresent()) {
+            group.setJmxExport(match.getJmxExport().get());
+        }
+    }
+
+    public static class ManagerSpec
+    {
+        private final List<ResourceGroupSpec> rootGroups;
+        private final List<SelectorSpec> selectors;
+
+        @JsonCreator
```

Comment:
It doesn't look like you're using JSON for any of the storage, so remove all these annotations. Also, it looks like these classes are copied from the file configuration manager, but have a parallel class hierarchy in `JdbcResourceGroupConfig`. I would consolidate them into a single set of structures.